src/components/calendar/bl-calendar.ts

```diff
@@ -11,309 +12,214 @@ import {
-
+  _calendarView: CalendarView = CALENDAR_VIEWS.DAYS;
   @state()
-  private _calendarYears: number[] = [];
-
+  _calendarYears: number[] = [];
   @state()
-  private _calendarDays: CalendarDay[] = [];
-
-  private _defaultValue: Date | Date[];
-
+  _calendarDays: CalendarDay[] = [];
   /**
-   * Defines the default selected date value for the calendar
+   * Fires when date selection changes
    */
-  @property({ type: Array, attribute: "default-value", reflect: true })
-  get defaultValue(): Date | Date[] {
-    return this._defaultValue;
-  }
-  set defaultValue(defaultValue) {
-    if (this.type === CALENDAR_TYPES.SINGLE && Array.isArray(defaultValue)) {
-      console.warn("Invalid prop value for defaultValue");
-    } else if (this.defaultValue) {
-      if (Array.isArray(this.defaultValue)) {
-        this._selectedDates = { ...this.defaultValue };
-      } else this._selectedDates = [this.defaultValue];
-    }
+  @event(blCalendarChangedEvent) _onBlCalendarChange: EventDispatcher<Date[]>;
+
+  static get styles(): CSSResultGroup {
+    return [style];
   }
+
   get months() {
-    return [...Array(12).keys()].map(month => {
-      return {
-        name: new Date(0, month + 1, 0).toLocaleString(this.locale, {
-          month: "long",
-        }),
-        value: month,
-      };
-    });
+    return [...Array(12).keys()].map(month => ({
+      name: new Date(0, month + 1, 0).toLocaleString(this.locale, { month: "long" }),
+      value: month,
+    }));
   }
+
   get days() {
-    return [...Array(7).keys()].map(day => {
-      return {
-        name: new Date(0, 0, day).toLocaleString(this.locale, { weekday: "short" }),
-        value: day,
-      };
-    });
-  }
-  /**
-   * Fires when date selection changes
-   */
-  @event("bl-calendar-change") private _onBlCalendarChange: EventDispatcher<Date[]>;
-  static get styles(): CSSResultGroup {
-    return [style];
+    return [...Array(7).keys()].map(day => ({
+      name: new Date(0, 0, day).toLocaleString(this.locale, { weekday: "short" }),
+      value: day,
+    }));
   }
+
+  public handleClearSelectedDates = () => {
+    this._selectedDates = [];
+    this._selectedRangeDates = { startDate: undefined, endDate: undefined };
+    this._onBlCalendarChange([]);
+    this.clearRangePickerStyles();
+  };
+
   getDayNumInAMonth(year: number, month: number) {
     return new Date(year, month + 1, 0).getDate();
   }
+
   getWeekDayOfDate(year: number, month: number) {
     return new Date(year, month, 1).getDay();
   }
+
   setPreviousCalendarView() {
     this.clearRangePickerStyles();
     if (this._calendarView === CALENDAR_VIEWS.DAYS) {
-      if (this._calendarMonth === FIRST_MONTH_INDEX) {
-        this._calendarMonth = LAST_MONTH_INDEX;
-        this._calendarYear -= 1;
-      } else this._calendarMonth -= 1;
+      this._calendarMonth === FIRST_MONTH_INDEX
+        ? ((this._calendarMonth = LAST_MONTH_INDEX), (this._calendarYear -= 1))
+        : (this._calendarMonth -= 1);
```

Comment:
I think we should not use short if where we cannot assign a variable.

(I saw it in a few more codes, I didn't want to write it in all of them.)